safe/gui/tools/minimum_needs/needs_calculator_dialog.py

```diff
@@ -45,7 +54,37 @@ def __init__(self, parent=None):
         self.setupUi(self)
         self.setWindowTitle(self.tr(
             'InaSAFE %s Minimum Needs Calculator' % get_version()))
-        self.polygon_layers_to_combo()
+
+        # get qgis map layer combobox object
+        self.layer = QgsMapLayerComboBox()
+        self.layer.setFilters(QgsMapLayerProxyModel.VectorLayer)
+
+        # get field that represent displaced count(population)
+        self.displaced = QgsFieldComboBox()
+        self.displaced.setFilters(QgsFieldProxyModel.Numeric)
+
+        # get field that represent aggregation name
+        self.aggregation_name = QgsFieldComboBox()
+        self.aggregation_name.setFilters(QgsFieldProxyModel.String)
+
+        # get field that represent aggregation id
+        self.aggregation_id = QgsFieldComboBox()
+
+        # add Qgis combo box to window
+        self.gridLayout_2.addWidget(self.layer, 0, 1)
```

Comment:
You can add this combobox in the UI file and not in the code.
In QTDesigner, use a QComboxBox and use the `promote` function.